src/aot/aot_main.cpp

```diff
@@ -132,8 +137,25 @@ int main(int argc, char* argv[])
         }
         break;
       default:
-        usage(std::cerr, argv[0]);
-        return 1;
+        auto unrecognized_opt = std::string(argv[optind - 1]);
+        if (unrecognized_opt.starts_with("--")) {
+          auto arg = unrecognized_opt.substr(2);
+          if (arg.find("=") != std::string::npos) {
+            auto split = util::split_string(arg,
+                                            '=',
+                                            /* remove_empty= */ true);
+            named_params[split[0]] = split[1];
+          } else {
+            named_params[arg] = "";
```

Comment:
Nit: you can distinguish between the string and no string by having the map be a `std::map<std::string, std::optional<std::string>>`. Then you would still have strict checking for `--foo` and `--foo=`.

Reply:
Ack. Though now it's valid to pass an empty string if `--foo` expects a string.